### ISSUE TITLE
Update 0.3.1

### ISSUE THREAD

**Issue by mhesselbarth** [r-spatialecology/landscapemetrics#70]
I think the bugfixes and changes (see attached `NEWS.md`), especially the one in `sample_lsm()` justify a CRAN realease (v.0.3.1). If you guys agree, I would submit it to CRAN later this week. 

Things to do:

- [x] Update @details of `list_lsm()` on how to get all metrics (d34573c5)
- [x] Update `DESCRIPTION` (Version: 0.3.1)
- [x] Update `cran-comments.md` (51098df3, 90896df8)
- [x] Update `NEWS.md` (97713fe1, e6a4bf1f, d34573c5, 6e9646ac, 90896df8)
- [x] Update docs/homepage (bbbd252e, 574fa425, 90896df8)
- [x] Run R CMD check and `devtools::check_win_release()` and `devtools::check_win_devel()` (5449333d)
- [ ] Merge with CRAN branch (#73)
- [ ] Create "release"-tag

**Comment by mhesselbarth** [comment]
I just fixed #71 and closed it (e6a4bf1f). 

Not sure what to do about the C++ issue at the moment. It seems like the devel version of R can't handle the TRUE/FALSE matrix for the neighbourhood in `get_adjacencies()`, right? But again, doesn't show up in any other check.

**Comment by Nowosad** [comment]
I think it could be due to a different C++ compilator (or a different version of a compilation) used by `devtools::check_win_devel()`.

**Comment by Nowosad** [comment]
@bitbacchus could you help us here?

**Comment by mhesselbarth** [comment]
Strange...I just downloaded the R development version and ran the `R CMD check` locally and no errors occurred.

``` 
R Under development (unstable) (2019-01-18 r75994) -- "Unsuffered Consequences"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

==> devtools::check(args = c('--as-cran'))

-- R CMD check results ----------------------------- landscapemetrics 0.3.1 ----
Duration: 5m 55.9s

0 errors v | 0 warnings v | 0 notes v

R CMD check succeeded
```

**Comment by mhesselbarth** [comment]
Also failing but with a different error (See [here](
https://builder.r-hub.io/status/landscapemetrics_0.3.1.tar.gz-8e969b2db1fe4b798331e4dda1032af0))

**Comment by marcosci** [comment]
It says status success there?

**Comment by mhesselbarth** [comment]
Yeah...but in the mail it says error and if you look at the report, there is also an error

**Comment by mhesselbarth** [comment]
If you guys agree (@Nowosad, @marcosci) I would submit the update to CRAN and see what happens...

**Comment by Nowosad** [comment]
Ok, I've checked it. It works;)

**Comment by marcosci** [comment]
But I think it will not pass because of the errors on devel, they check the same on CRAN.

Could it be that it's because of the check we are doing there for the input of the neighborhood definition? If I remember correctly, the devel version of R checks for logical operators now and throws an error if wrongly used (I think we only want a single ampersand there, or?).

I just sent out a new version of landscapemetrics to win_devel, let's at least wait for that :-)

**Comment by marcosci** [comment]
No, that's not the problem either.
I fixed it anyways by skipping the example on CRAN.

We should try to find a better solution for create_neighborhood ... right now we code NA, 0, 1 in the filter matrix, but the 0 is actually unnecessary? The center of the filter should always be the middle of the matrix, so endocing as true/false or 1/0 should be enough, or?

**Comment by Nowosad** [comment]
Maybe this info could be helpful (source: https://github.com/ropensci/drake/releases)

> Version 6.2.1 is a hotfix to address the failing automated CRAN checks for 6.2.0. Chiefly, in CRAN's Debian R-devel (2018-12-10) check platform, errors of the form "length > 1 in coercion to logical" occurred when either argument to && or || was not of length 1 (e.g. nzchar(letters) && length(letters)).

**Comment by marcosci** [comment]
I fixed that in one of the last commits, but there was still an error with rcpp_create_neighborhood (and the conversersion there of logical values, hence my last comment) :+1: .

**Comment by Nowosad** [comment]
@marcosci What's the error message now?

**Comment by marcosci** [comment]
Yes, because the center (we now code it as a 0) is anyways always on the middle, or?
Would also make it it easier to create a filter matrix.

**Comment by Nowosad** [comment]
That's all true. However, now we can have situation with the center in every location. Maybe it is not useful now, but it could be someday... Cannot we think of a solution on the Rcpp level?

**Comment by marcosci** [comment]
Most likely - yes :yum: 

I can check that next week, when I am back in the office - I don't what happens with the NAs in a matrix with RcppArmadillo and need the computer in the office for that.

**Comment by mhesselbarth** [comment]
Okay...I think I fixed the error finally (5449333d). 

The problem was that the test if  `neighbourhood` is either 4, 8 or a matrix didn't really work. If `neighbourhood` was a matrix, `neighbourhood != 4` checked if each element in the matrix unequals four (and not if the object itself unequals 4).

```
> neighbourhood <- diagonal_matrix <- matrix(c(1,  NA,  1, NA,  0, NA, 1,  NA,  1), 3, 3, byrow = TRUE)

> neighbourhood != 4
     [,1] [,2] [,3]
[1,] TRUE   NA TRUE
[2,]   NA TRUE   NA
[3,] TRUE   NA TRUE

> !identical(neighbourhood, 4)
[1] TRUE
```

**Comment by mhesselbarth** [comment]
I just submitted to CRAN. @Nowosad Maybe you can open a new issue about the tests? I would close this one
